tests/cpp/helpers.cu

```diff
@@ -40,4 +47,14 @@ std::shared_ptr<DMatrix> RandomDataGenerator::GenerateDeviceDMatrix(bool with_la
       0, bins_);
   return m;
 }
+
+void SetUpRMMResource() {
+#if defined(XGBOOST_USE_RMM) && XGBOOST_USE_RMM == 1
+  using cuda_mr_t = rmm::mr::cuda_memory_resource;
+  using pool_mr_t = rmm::mr::pool_memory_resource<cuda_mr_t>;
+  auto cuda_mr = std::make_unique<cuda_mr_t>();
+  auto pool_mr = std::make_unique<pool_mr_t>(cuda_mr.release());
+	rmm::mr::set_default_resource(pool_mr.release());
```

Comment:
These will technically leak when the application exits since the resources are never destroyed.

Reply:
This is no longer the case with the latest revision. https://github.com/dmlc/xgboost/blob/78c22544d89306d5f4b24ea19278b40b5e181b87/tests/cpp/helpers.cu#L63-L66